llap-server/src/java/org/apache/hadoop/hive/llap/cache/LowLevelLrfuCachePolicy.java

```diff
@@ -491,29 +499,32 @@ private int moveMinChildUp(int targetPos, long time, double comparePri) {
     }
     if (leftPri <= rightPri) { // prefer left, cause right might be missing
       heap[targetPos] = left;
-      left.indexInHeap = targetPos;
+      LrfuCacheAttribute leftCacheAttribute = (LrfuCacheAttribute) left.cacheAttribute;
+      leftCacheAttribute.indexInHeap = targetPos;
```

Comment:
why doesn't this use a setter?

Reply:
less work on the compiler :D i can use setter if you think it is better.

Reply:
would hope that the compiler is optimized to inline the setter code. Using setter makes it better encapsulated.